drivers/ethernet/eth_xmc4xxx.c

```diff
@@ -799,6 +801,31 @@ static inline int eth_xmc4xxx_init_timestamp_control_reg(ETH_GLOBAL_TypeDef *reg
 	return 0;
 }
 
+static int eth_xmc4xxx_get_mac_address_eeprom(const struct device *dev)
```

Comment:
Instead of putting this in the drivers, wouldn't make sense to put this on a higher level and use `eth_xmc4xxx_set_config` to apply?

Reply:
+ 1

maybe we can have some generic implementation for setting the mac address in 
in `net_if_post_init()` just before bringing the iface up of `subsys/net/ip/net_if.c`. 

What do you think? @jukkar @rlubos

Reply:
should be after iface init, but before iface up